circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/conf/CircusTrainTransformOptions.java

```diff
@@ -15,26 +15,20 @@
  */
 package com.hotels.bdp.circustrain.core.conf;
 
-import static com.hotels.bdp.circustrain.core.conf.TableParametersConfig.TABLE_REPLICATION_TABLE_PARAMETERS;
-
-import java.util.Map;
+import static com.hotels.bdp.circustrain.core.conf.CircusTrainTransformOptions.TRANSFORM_OPTIONS_BEAN;
+import static com.hotels.bdp.circustrain.core.conf.CircusTrainTransformOptions.TRANSFORM_OPTIONS_PROPERTY;
 
 import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.context.annotation.Configuration;
 
-@Configuration
-@ConfigurationProperties(prefix = TABLE_REPLICATION_TABLE_PARAMETERS)
-public class TableParametersConfig {
-
-  public static final String TABLE_REPLICATION_TABLE_PARAMETERS = "table-properties";
+import com.hotels.bdp.circustrain.api.conf.TransformOptions;
 
-  private Map<String, String> parameters;
+@Configuration(TRANSFORM_OPTIONS_BEAN)
+@ConfigurationProperties(TRANSFORM_OPTIONS_PROPERTY)
+public class CircusTrainTransformOptions extends TransformOptions {
 
-  public Map<String, String> getParameters() {
-    return parameters;
-  }
+  public static final String TRANSFORM_OPTIONS_BEAN = "transformOptions";
+  public static final String TRANSFORM_OPTIONS_PROPERTY = "transform-options";
+  public static final String TABLE_REPLICATION_TABLE_PARAMETERS = "table-properties";
```

Comment:
Should we not just call this `TABLE_PROPERTIES` here and only refer to them as "params" when they get closer to Hive? I remember us agreeing on some convention here but can't remember exactly where the boundary was in property vs param.

Reply:
I think everything everything external is a property, and internal is a param, and the boundary is as close to the user as possible. So yep I'll do `TABLE_REPLICATION_TABLE_PARAMETERS` -> `TABLE_PROPERTIES`, and then everything which uses this will be called a param.

Reply:
:+1: